lib/blockscore/collection.rb

```diff
@@ -133,8 +133,7 @@ def create(params = {})
     # @api public
     def retrieve(id)
       each do |item|
-        next unless item.id == id
-        return item
+        return item if item.id.eql?(id)
```

Comment:
None of the id's in the system are integers, they are strings (similar to uuid's). So eql? is the best I know - correct?